### PR TITLE
add SetBlockMeta warning in documentation

### LINKED ISSUE #4058: Door doesn't close when lever is turned off from a plugin
Client version:   1.12.2
Server OS:        Linux
Commit id:        76a0971279de47c5a1d8510848e88e57cc6eed72

### Expected behavior

Door closes when lever is turned off from a plugin.

### Actual behavior

Door doesn't close when lever is turned off from a plugin.

### Steps to reproduce the behavior

1. Place a door with next to it a block with a lever on it.
2. Turn the lever on by right-clicking.
3. From a plugin, call `SetBlockMeta` on the world with the location of the lever, and one of the following metas (for example `cRoot:Get():GetWorld("world"):SetBlockMeta(-1, 7, 9, 4, false, true)`):

Meta (+8 for on) | Description
-- | --
0 | Lever on block bottom points east when off
1 | Lever on block side facing east
2 | Lever on block side facing west
3 | Lever on block side facing south
4 | Lever on block side facing north
5 | Lever on block top points south when off
6 | Lever on block top points east when off
7 | Lever on block bottom points south when off

Update: the door closes when I dig the unlit redstone torch next to the lever, but it doesn't open again when I place a redstone torch next to the lever. When I turn the lever on or off as a player, the door just closes or opens fine, even if there's an unlit redstone torch next to it.

Workaround: also `SetBlockMeta` on the door.

My build looks like the following:

Screenshot

### PR DESCRIPTION
**Pull request by pwnorbitals**
Fix #4058. 
Feedback appreciated

### CODE REVIEW

**Comment by bearbin** [comment]
I don't think that 'generate a simulator update' is the most understandable wording for less-knowledgeable plugin writers. I think adding examples for the current simulators here would be useful.

File: Server/Plugins/APIDump/APIDesc.lua [hunk 1]
```diff
@@ -1142,7 +1142,7 @@ return
 							Type = "number",
 						},
 					},
-					Notes = "Sets the block meta at the specified relative coords",
+					Notes = "Sets the block meta at the specified relative coords. Any call to SetBlockMeta will not generate a simulator update (water, lava, redstone), consider using SetBlock instead.",
```
Comment (tigerw) [inline Server/Plugins/APIDump/APIDesc.lua:1145]: This is the API for cChunkDesc right? That's just a container for block data during World generation, no setters will generate any simulator updates.

File: Server/Plugins/APIDump/Classes/BlockArea.lua [hunk 2]
```diff
@@ -1621,7 +1621,7 @@ return
 						Type = "number",
 					},
 				},
-				Notes = "Sets the block meta at the specified absolute coords",
+				Notes = "Sets the block meta at the specified absolute coords. Any call to SetBlockMeta will not generate a simulator update (water, lava, redstone), consider using SetBlock instead.",
```
Comment (tigerw) [inline Server/Plugins/APIDump/Classes/BlockArea.lua:1624]: Similarly for cBlockArea, it doesn't do any simulator updates at all (for speed when setting a large area).

**Comment by pwnorbitals** [comment]
Is it documented somewhere that cBlockArea and cChunkDesc never generate simulator updates ?

**Comment by tigerw** [comment]
Implicitly for cChunkDesc: https://api.cuberite.org/cChunkDesc.html
Not really for cBlockArea: https://api.cuberite.org/cBlockArea.html

**Review by tigerw — APPROVED** [review]

**Pull request merged**